src/Dofs/ConstraintHandler.jl

```diff
@@ -818,81 +816,62 @@ end
 
-function add!(ch::ConstraintHandler, fh::FieldHandler, dbc::Dirichlet; warn_not_in_cellset=true)
-    if warn_not_in_cellset && !(_in_cellset(ch.dh.grid, fh.cellset, dbc.faces; all=true))
-        @warn("You are trying to add a constraint a face/edge/node that is not in the cellset of the fieldhandler. This location will be skipped")
-    end
-
-    celltype = getcelltype(ch.dh.grid, first(fh.cellset)) #Assume same celltype of all cells in fh.cellset
-
-    # Extract stuff for the field
-    field_idx = find_field(fh, dbc.field_name)
-    interpolation = getfieldinterpolation(fh, field_idx)
-    field_dim = getfielddim(fh, field_idx)
-
-    if !all(c -> 0 < c <= field_dim, dbc.components)
-        error("components $(dbc.components) not within range of field :$(dbc.field_name) ($(field_dim) dimension(s))")
-    end
-
-    # Empty components means constrain them all
-    isempty(dbc.components) && append!(dbc.components, 1:field_dim)
-
-    if eltype(dbc.faces)==Int #Special case when dbc.faces is a nodeset
-        bcvalue = BCValues(interpolation, default_interpolation(celltype), FaceIndex) #Not used by node bcs, but still have to pass it as an argument
-    else
-        bcvalue = BCValues(interpolation, default_interpolation(celltype), eltype(dbc.faces))
-    end
-
-    Ferrite._add!(ch, dbc, dbc.faces, interpolation, field_dim, field_offset(fh, dbc.field_name), bcvalue, fh.cellset)
-    return ch
-end
-
-# If all==true, return true only if all items in faceset/nodeset are in the cellset
-# If all==false, return true if some items in faceset/nodeset are in the cellset
-function _in_cellset(::AbstractGrid, cellset::Set{Int}, faceset::Set{<:BoundaryIndex}; all=true)
-    for (cellid,faceid) in faceset
-        if cellid in cellset
-            all || return true
-        else
-            all && return false
-        end
+# Return the intersection of the FieldHandler set and the Dirichlet BC set
+function filter_dbc_set(::AbstractGrid, fhset::AbstractSet{Int}, dbcset::AbstractSet{<:BoundaryIndex})
+    r = empty(dbcset)::typeof(dbcset)
+    for x in dbcset
+        cellid, _ = x
+        cellid in fhset && push!(r, x)
     end
-    return all # if not returned by now and all==false, then no `cellid`s where in cellset
+    return r
 end
-
-function _in_cellset(grid::AbstractGrid, cellset::Set{Int}, nodeset::Set{Int}; all=true)
-    nodes = Set{Int}()
-    for cellid in cellset
-        for nodeid in grid.cells[cellid].nodes
-            nodeid ∈ nodes || push!(nodes, nodeid)
-        end
+function filter_dbc_set(grid::AbstractGrid, fhset::AbstractSet{Int}, dbcset::AbstractSet{Int})
+    r = empty(dbcset)
+    ns = Set{Int}()
+    for cc in CellIterator(grid, fhset, UpdateFlags(;nodes=true, coords=false))
+        union!(ns, cc.nodes)
     end
-
-    for nodeid in nodeset
-        if nodeid ∈ nodes
-            all || return true 
-        else
-            all && return false
-        end
+    for x in dbcset
+        x in ns && push!(r, x)
```

Comment:
```suggestion
    for nodeid in dbcset
        nodeid in ns && push!(r, nodeid)
```
(x confused me thinking of coordinate:))